crates/oxc_linter/src/rules/unicorn/consistent_function_scoping.rs

```diff
@@ -260,6 +260,21 @@ impl<'a> Visit<'a> for ReferencesFinder {
         self.references.push(it.reference_id().unwrap());
     }
 
+    fn visit_jsx_element_name(&mut self, it: &oxc_ast::ast::JSXElementName<'a>) {
+        if !matches!(it, oxc_ast::ast::JSXElementName::IdentifierReference(_)) {
+            walk::walk_jsx_element_name(self, it);
+        }
+    }
+
+    fn visit_jsx_member_expression_object(
+        &mut self,
+        it: &oxc_ast::ast::JSXMemberExpressionObject<'a>,
+    ) {
+        if !matches!(it, oxc_ast::ast::JSXMemberExpressionObject::IdentifierReference(_)) {
+            walk::walk_jsx_member_expression_object(self, it);
+        }
+    }
+
```

Comment:
This test failed, and I added the above code to work around it, maybe you know of a better solution. @camc314 

https://github.com/oxc-project/oxc/blob/4b7e518759a9cb6edf24b99ee4b9d5532bb64942/crates/oxc_linter/src/rules/unicorn/consistent_function_scoping.rs#L603-L611

Reply:
as in that test started passing? instead of reporting a diagnostic?

Reply:
Yes

Reply:
I have some thoughts about this. I'll open a separate issue.